service/src/main/java/uk/gov/hmcts/reform/fpl/service/CaseSubmissionValidatorService.java

```diff
@@ -0,0 +1,57 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import com.google.common.collect.ImmutableList;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.fpl.enums.SectionType;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import javax.validation.ConstraintViolation;
+import javax.validation.Validation;
+import javax.validation.Validator;
+
+import static uk.gov.hmcts.reform.fpl.enums.SectionType.APPLICANT;
+import static uk.gov.hmcts.reform.fpl.enums.SectionType.CASENAME;
+import static uk.gov.hmcts.reform.fpl.enums.SectionType.CHILDREN;
+import static uk.gov.hmcts.reform.fpl.enums.SectionType.DOCUMENTS;
+import static uk.gov.hmcts.reform.fpl.enums.SectionType.GROUNDS;
+import static uk.gov.hmcts.reform.fpl.enums.SectionType.HEARING;
+import static uk.gov.hmcts.reform.fpl.enums.SectionType.ORDERS;
+
+@Service
+public class CaseSubmissionValidatorService {
+
+    public List<String> validateCaseDetails(CaseData caseData) {
+        ImmutableList.Builder<String> caseErrors = ImmutableList.builder();
+
+        Validator validator = Validation.buildDefaultValidatorFactory().getValidator();
```

Comment:
New validator should not be created for every validation. Its should be class level field or injected ideally.